source/manual/intro-to-docker.html.md

```diff
@@ -0,0 +1,331 @@
+
+### Run a container from an image
+
+* Make sure you have docker running:
+
+```shell
+$mac docker --version
+Docker version 18.09.2, build 6247962
+```
+
+#### What are Docker images and a container?
+
+* Run from the root of the [content-publisher] project:
+
+```shell
+$mac docker run -it --rm ruby:2.6.3 bash
+```
+
+`ruby` is the "image" that we specify. Using object oriented programming as an analogy, an image is like a "class". When we "instantiate" an instance of a class, these are called "containers".
+
+The `ruby` image we are using is an ubuntu system with Ruby pre-installed. Images are downloaded from the Docker Registry. We are also specifying that we want the image that provides Ruby 2.6.3.
+
+The `bash` argument at the end is passed to the docker run command. It will execute this inside the container, which means we get a bash shell.
+
+What are the flags?
+
+```shell
+$mac docker run --help
+...
+-i, --interactive                    Keep STDIN open even if not attached
+...
+-t, --tty                            Allocate a pseudo-TTY
+...
+    --rm                             Automatically remove the container when it exits
+```
+
+### Volumes
+
+* Mount your current directory as a volume to the container by running:
+
+```shell
+$mac docker run -it --rm -v $(pwd):/app ruby:2.6.3 bash
+```
+
+This will map your current directory (the root of the [content-publisher] project) to `/app` inside of the container. So now the files on your `$mac` for the [content-publisher] are now also available inside of the container.
+
+* If you were to run bundle install for the app:
+
+```shell
+$dev cd /app
+$dev bundle install
+```
+
+You can see that the gems required for content-publisher are installed! However, anything you do here won't persist - if you were to quit the container and then go back in exactly the same way, all the gems would need to be re-installed again.
+
+By default gems are saved to `/usr/local/bundle` within the container. But everything in the container is destroyed and reset to the image when it's shut down.
+
+We could mount that path to the same on your `$mac` but there is a Docker way of providing storage.
+
+#### Persistent volumes
+
+We don't want to be re-installing and doing setup every time we quit and re-enter a container. Docker allows you to create separate volumes for persistent data.
+
+* Create a persistent volume for our gems:
+
+```shell
+$mac docker volume create content-publisher-bundle
+```
+
+*   Run docker with that volume by using the `-v` flag and passing it the name of the volume:
+
+```shell
+$mac docker run -it --rm -v $(pwd):/app -v `content-publisher-bundle`:/usr/local/bundle ruby:2.6.3 bash
+```
+
+*   Then install the gems again:
+
+```shell
+$dev cd /app
+$dev bundle install
+```
+
+This time the gems will install on the `content-publisher-bundle` which is mapped to the container at the `/usr/local/bundle` path.
+
+### Specifying dependencies (node, PostgreSQL, Chrome)
```

Comment:
I think there are really two separate things going on here, as per my comment below. One is about networking, and the other is about images/Dockerfiles. Since the section is quite long, it might also help the reader to split it up - we could have a 'Images (Revisited)' section or something to continue with installing Chrome.